l2-contracts/contracts/ConsensusRegistry.sol

```diff
@@ -0,0 +1,278 @@
+
+/// @title ConsensusRegistry
+/// @dev A contract to manage consensus nodes and committees.
+contract ConsensusRegistry {
+    // Owner of the contract (i.e., the governance contract).
+    address public owner;
+    // An array to keep track of node owners.
+    address[] public nodeOwners;
+    // A map of node owners => nodes
+    mapping(address => Node) public nodes;
+    // The current validator committee list.
+    CommitteeValidator[] public validatorCommittee;
+    // The current attester committee list.
+    CommitteeAttester[] public attesterCommittee;
+
+    /// @dev Represents a consensus node.
+    struct Node {
+        // A flag stating if the node is inactive.
+        // Inactive nodes are not considered when selecting committees.
+        bool isInactive;
+        // Validator's voting weight.
+        uint256 validatorWeight;
+        // Validator's BLS12-381 public key.
+        bytes validatorPubKey;
+        // Validator's Proof-of-possession (a signature over the public key).
+        bytes validatorPoP;
+        // Attester's Voting weight.
+        uint256 attesterWeight;
+        // Attester's ECDSA public key.
+        bytes attesterPubKey;
+    }
+
+    /// @dev Represents a validator committee member.
+    struct CommitteeValidator {
+        address nodeOwner;
+        uint256 weight;
+        bytes pubKey;
+        bytes pop;
+    }
+
+    /// @dev Represents an attester committee member.
+    struct CommitteeAttester {
+        uint256 weight;
+        address nodeOwner;
+        bytes pubKey;
+    }
+
+    error UnauthorizedOnlyOwner();
+    error UnauthorizedOnlyOwnerOrNodeOwner();
+    error NodeOwnerAlreadyExists();
+    error NodeOwnerDoesNotExist();
+    error NodeOwnerNotFound();
+    error ValidatorPubKeyAlreadyExists();
+    error AttesterPubKeyAlreadyExists();
+
+    modifier onlyOwner() {
+        if (msg.sender != owner) {
+            revert UnauthorizedOnlyOwner();
+        }
+        _;
+    }
+
+    modifier onlyOwnerOrNodeOwner(address nodeOwner) {
+        if (msg.sender != owner && msg.sender != nodeOwner) {
+            revert UnauthorizedOnlyOwnerOrNodeOwner();
+        }
+        _;
+    }
+
+    constructor(address _owner) {
+        owner = _owner;
+    }
+
+    /// @notice Adds a new node to the registry.
+    /// @dev Fails if node owner already exists.
+    /// @dev Fails if a validator with the same public key already exists.
+    /// @dev Fails if an attester with the same public key already exists.
+
+    /// @param nodeOwner The address of the new node's owner.
+    /// @param validatorWeight The voting weight of the validator.
+    /// @param validatorPubKey The BLS12-381 public key of the validator.
+    /// @param validatorPoP The proof-of-possession (PoP) of the validator's public key.
+    /// @param attesterWeight The voting weight of the attester.
+    /// @param attesterPubKey The ECDSA public key of the attester.
+    function add(
+        address nodeOwner,
```

Comment:
We always name function input variables starting with `_` prefix

Reply:
Fixed here: https://github.com/matter-labs/era-contracts/pull/555/commits/ca48a3028dcdffb8c5228fe8973dae3acd9f10c4